### DISCUSSION TITLE
Can I use 4.1.7 with my new 1.5.8a install?

### DISCUSSION THREAD

**Discussion by DanTomlinson** [lat9/edit_orders#234] [Q&A]
Hi all!
I've been running 4.1.7 with a couple of minor modifications, and with the new installation of ZC1.5.8a figured
it's time to upgrade EO also.  It WON'T behave!
We use EO to modify shipments if needed - adding an item, changing a quantity, dropping an item etc.
The "new" with the "replace everything" method significantly corrupts orders of the past!

- changing an order quantity recalculates the cart to "today's" prices.  Instead of what they were ordered at BEFORE we started a sale.  Or after a different sale ended!  
- Changing quantity to 0 drops the item instead of showing that we are shipping 0 from the 6 ordered.
- It deletes the original orders_products items instead of updating them - losing other info also that was stored with the original order (what we had in stock when the order was placed, whether we had to order to fulfill the order etc etc)

So - Can I run the 4.1.7 version on 1.5.8a?  What is the known breakage?

Thanks!
Dan
BeadFX

**Comment by lat9** [comment]
Dan, you should use the "manual" pricing for orders' edits.  That will keep the pricing as present in the original order, if not manually-overridden.  EO 4.1.7 was released in 2016, so I've got no idea how it will interoperate with zc158a (nor do I support that interoperation).

**Comment by DanTomlinson** [comment]
Thanks for that piece Cindy.  Yes, editing an order to shippable quantities
after a sale ends and having all the prices increase would be ... annoying
for the customers!

Any chance that you'll go back to updating-in-place instead of deleting the
whole order and replacing some of it?

Can you write a 0 quantity item using the current method?

2 pieces on this.
 - we want the original order in place so customers can see what "didn't
ship" also.
 - I have 5 additional fields in the orders_products table (an updated
classes/shopping_cart and classes/order that support them.
   (I don't use "options"!)
  -1  base_price - the product price with no discounts applied
  -2  original_quantity - how the order was originally placed
These 2 fields allow giving information to the customer to show changes to
the order as well as
  calculate and show them the discount on the final invoice.

3 more fields are for our "over-selling" feature - selling more than in the
actual store.  80% we can have next day.
  -3  otf_req - The cart know we don't have enough in the store, and places
with assistance an order from a wholesaler
  -4  otf_oh -  How many we actually/should have in the store at the time
the order is placeed
  -5  otf_qn -  How many the supplier shows in their inventory available to
us.  (We get direct stock feeds from 3 wholesalers)

These need to carry over with the order data.

See the attached with screenshots of how they are used.

I'm working with  4.7.0 a/o Feb 27 - so I know I'm being  cautiously
brave.



I can sort-of see where to add the fields to "re-write" to ORDERS_PRODUCTS
at around line 572 in extra_functions/edit_orders_functions.php,
but where do I get the original numbers from the original ORDERS_PRODUCTS
lines?

Since I don't want to "remove a product" but just change ship quantity to
0, can I just modify the query at around line 876 or just "disable" the
delete?

(I don't really care about stock quantity updates - I'll fix that every
couple hours with a sync script from the in-store system anyway)


Your help is absolutely appreciated!!!!

Thanks,
Dan Tomlinson
Toronto


PS - I know you'll want to see "your"  Bootstrap template perform.  I went
live with it on Sunday after modifications and testing.
So far I'm SO impressed with the speed and a very consistent look&feel -
not all the features enabled yet tho.
Made for "Mobile First" and 100% consistent look between desktop and phone.
Replaced my previous use of "Dynamic Filters" with a scratch re-write
bypassing "Read-only" attributes to a parallel filter table.

Love to hear your thoughts on that too!

Thanks, and chat soon!
Dan

**Comment by DanTomlinson** [comment]
Sorry - forgot to add the URL

www.beadfx.com/catalogue

Dan
